inbm/dispatcher-agent/dispatcher/dispatcher_class.py

```diff
@@ -141,6 +142,20 @@ def start(self, tls: bool = True) -> None:
         with ota_lock:
             self._perform_startup_tasks()
 
+        # Run scheduler to schedule the task during startup.
+        scheduler = APScheduler()
+        sqliteManager = SqliteManager()
+        single_schedule_list = sqliteManager.get_all_single_schedule_in_priority()
+        logger.info(f"Total single scheduled tasks: {len(single_schedule_list)}")
+        for single_schedule in single_schedule_list:
+            scheduler.add_single_schedule_job(self.do_install, single_schedule)
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Refactoring$</h2>**
**Suggestion:** Refactor the scheduling logic into a separate method to improve code readability and maintainability.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        def schedule_tasks(self):
            scheduler = APScheduler()
            sqliteManager = SqliteManager()
            single_schedule_list = sqliteManager.get_all_single_schedule_in_priority()
            logger.info(f"Total single scheduled tasks: {len(single_schedule_list)}")
            for single_schedule in single_schedule_list:
                scheduler.add_single_schedule_job(self.do_install, single_schedule)
        schedule_tasks(self)
```